libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -662,6 +666,17 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		solAssert(!functionDef || functionDef->isImplemented(), "");
 
 		vector<string> args;
+		if (memberAccess && functionType->bound())
+		{
+			solAssert(functionDef->parameters().size() == arguments.size() + 1, "");
+			if (functionType->takesArbitraryParameters())
+				args += IRVariable(memberAccess->expression()).stackSlots();
+			else
+				args += convert(memberAccess->expression(), *functionDef->parameters()[0]->type()).stackSlots();
```

Comment:
I don't think the case with `takesArbitraryParameters()` is possible. AFAIK internal and external functions cannot have arbitrary parameters. Only the special functions like the `abi` helpers can but here we know we're dealing with an internal function.

I kept it mostly because the loop below had it too so maybe it's possible after all. But no tests fail when I replace it with an assert:
```c++
solAssert(!functionType->takesArbitraryParameters(), "");
```

Reply:
I think it's better to add the assert here.

Reply:
ok. I put it in a separate PR: #8997